src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -191,6 +189,16 @@ public String getName() {
     return components.values();
   }
 
+  /**
+   * Returns a mapping of exercise IDs to modules. The keys are exercise IDs, and the values are
+   * maps from language codes to module names. Note, that some exercises use modules that are not in
+   * the course configuration file, so the modules may not be in {@link Course#getModules}.
+   */
+  @NotNull
+  public Map<Long, Map<String, String>> getExerciseModules() {
```

Comment:
the nested map is the only thing that has drawn attention. maybe so multiMap cab be used, but this is so optional :+1: